conformance_tests/test_tiovx/test_ext_supplementary_data.c

```diff
@@ -0,0 +1,1435 @@
+            break;
+        case VX_TYPE_IMAGE:
+            ref = (vx_reference)vxCreateImage(context, 16, 16, VX_DF_IMAGE_U8);
+            break;
+        case VX_TYPE_LUT:
+            ref = (vx_reference)vxCreateLUT(context, VX_TYPE_UINT8, 4);
+            break;
+        case VX_TYPE_MATRIX:
+            ref = (vx_reference)vxCreateMatrix(context, VX_TYPE_INT32, 3, 3);
+            break;
+        case VX_TYPE_OBJECT_ARRAY:
+        {
+            vx_image image = vxCreateImage(context, 16, 16, VX_DF_IMAGE_U8);
+            ref = (vx_reference)vxCreateObjectArray(context, (vx_reference)image, 4);
+            vxReleaseImage(&image);
+            break;
+        }
+        case VX_TYPE_PYRAMID:
+            ref = (vx_reference)vxCreatePyramid(context, 2, VX_SCALE_PYRAMID_HALF, 16, 16, VX_DF_IMAGE_U8);
+            break;
+        case VX_TYPE_SCALAR:
+        {
+            vx_uint32 value = 0;
+            ref = (vx_reference)vxCreateScalar(context, VX_TYPE_UINT32, &value);
+            break;
+        }
+        case VX_TYPE_TENSOR:
+        {
+            vx_size dims[3] = {4, 4, 4};
+            ref = (vx_reference)vxCreateTensor(context, 3, dims, VX_TYPE_INT32, 0);
+            break;
+        }
+        case VX_TYPE_USER_DATA_OBJECT:
+        {
+            user_data_t user_data = {0};
+            ref = (vx_reference)vxCreateUserDataObject(context, "user_data_t", sizeof(user_data_t), &user_data);
+            break;
+        }
+        case VX_TYPE_REMAP:
+            ref = (vx_reference)vxCreateRemap(context, 16, 16, 16, 16);
+            break;
+        case VX_TYPE_THRESHOLD:
+            ref = (vx_reference)vxCreateThreshold(context, VX_THRESHOLD_TYPE_BINARY, VX_TYPE_UINT8); /* This will need updating!!)*/
+            break;
+        case TIVX_TYPE_RAW_IMAGE:
+            {
+                tivx_raw_image_create_params_t params = {
+                    .width = 16,
+                    .height = 16,
+                    .line_interleaved = vx_false_e,
+                    .meta_height_before = 1,
+                    .meta_height_after = 1,
+                    .num_exposures = 2,
+                    .format = {
+                        {.msb = 7, .pixel_container = TIVX_RAW_IMAGE_8_BIT},
+                        {.msb = 7, .pixel_container = TIVX_RAW_IMAGE_8_BIT},
+                        {.msb = 7, .pixel_container = TIVX_RAW_IMAGE_8_BIT}
+                    }
+                };
+                ref = (vx_reference)tivxCreateRawImage(context, &params);
+                break;
+            }
+        default:
+            printf("ERROR in TEST! Found unknown type %d (%s)\n", type, enumToString(type));
+    }
+    return ref;
+}
+
+TEST (supplementary_data, testRefCount)
+{
+    vx_context context = context_->vx_context_;
+
+    user_data_t user_data = {.numbers = {1, 2, 3, 4}};
+    vx_user_data_object exemplar = vxCreateUserDataObject(context, "user_data_t", sizeof(user_data_t), &user_data);
```

Comment:
Can we please add an ASSERT_VX_OBJECT here to ensure this was created properly?  It looks like there are a few instances of this throughout the file

Reply:
done